crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2289,7 +2289,16 @@ impl<'db> TypeInferenceBuilder<'db> {
                                 // (`Type::format`?) that handles the `__format__` method.
                                 // Conversion flags should be handled before calling `__format__`.
                                 // https://docs.python.org/3/library/string.html#format-string-syntax
-                                if !conversion.is_none() || format_spec.is_some() {
+                                if !conversion.is_none() {
+                                    collector.add_expression();
```

Comment:
Isn't it necessary to infer the expression type if `conversion` is some, e.g for `f"{10!s:abcd}`?

Reply:
In that case, `conversion` will change to `ConversionFlag::Str`, but this enum has no subexpressions or similar. Not that I'm not adding proper type inference support for these f-strings here, I'm merely trying to make sure we infer types for all subexpressions, so that we do not panic. So it's possible we need to look at this flag once we implement this properly, but I don't see why it would be needed now.

Reply:
What I understand is that we only take the branch inferring the expression types if the conversion is none. What if it isn't?

Reply:
You are totally right. Fixed & added a proper test.